s3.py

```diff
@@ -161,29 +241,69 @@ def __init__(self, integration, client_class=None):
         for setting in integration.settings:
             if setting.key.endswith('_bucket'):
                 self.buckets[setting.key] = setting.value
+
+    def _generate_s3_url(self, bucket, path):
+        """Generates an S3 URL
+
+        :param bucket: Bucket name
+        :type bucket: string
+
+        :return: S3 URL
+        :rtype: string
+        """
+        key = path
+
+        # NOTE: path can be an empty string meaning that
+        # we need to generate a URL pointing at the root directory of the bucket.
+        # However, boto3 doesn't allow us to pass the key as an empty string.
+        # As a workaround we set it to a dummy string and later remove it from the generated URL
+        if not path:
+            key = 'dummy'
```

Comment:
I get why this is necessary but I'd feel better if this dummy key included a frivolous Unicode character, like `dummy☃`. I could see someone creating a bucket actually called `dummy` for testing purposes.

Reply:
I guess it doesn't matter too much since you're never sending `dummy` to the server, so a preexisting bucket by that name won't cause a problem.